### PR TITLE
New: Use Testim.io as part of a CI process [ED-4051]

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: .github/scripts/testim_ci.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      "tests-cli",
+      `"wp config set WP_HOME ${tunnelUrl}"`,
+    ]),
+    runProgram('npx', [
+      "wp-env",
+      "run",
+      "tests-cli",
+      `"wp config set WP_SITEURL ${tunnelUrl}"`,
+    ]),
+  ]).catch(e => console.error(e));
```
Comment (ronkelementor) [inline .github/scripts/testim_ci.js:44]: need to stop the process if you catch error here

File: .github/workflows/testim-ci.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install dependencies
+        run: npm install
```
Comment (ronkelementor) [inline .github/workflows/testim-ci.yml:31]:
```suggestion
        run: npm ci
```

File: .github/workflows/testim-ci.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          WP_CORE_VERSION: ${{ matrix.wpCoreVersion }}
+        run: node ./.github/scripts/build-wp-env.js
+      - name: Install wp-env
+        run: npm install --no-package-lock --no-save @wordpress/env@4.0.0
+      - name: Install WordPress environment
+        run: |
+          npx wp-env start
+          # change wp-env folder owner to www-data
+          sudo chown 33:33 -R ~/wp-env/
+          npx wp-env run tests-cli "bash -c 'bash elementor-config/import-temapltes.sh'"
```
Comment (ronkelementor) [inline .github/workflows/testim-ci.yml:75]: do you need the templates here?

File: .github/workflows/testim-ci.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      - name: WordPress debug information
+        run: |
+          npx wp-env run tests-cli "wp core version"
+          npx wp-env run tests-cli "wp --info"
+      - name: Install Testim
+        run: npm install --no-package-lock --no-save @testim/testim-cli
+      - name: Install Ngrok
+        run: npm install --no-package-lock --no-save ngrok
```
Comment (ronkelementor) [inline .github/workflows/testim-ci.yml:83]:
```suggestion
        run: npm install --no-package-lock --no-save ngrok@4.0.1
```

File: .github/workflows/testim-ci.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+on:
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  build-plugin:
+    name: Build plugin
+    runs-on: ubuntu-latest
+    if: github.event.review.state == 'approved'
```
Comment (ronkelementor) [inline .github/workflows/testim-ci.yml:11]:
```suggestion
    if: github.event.review.state == 'approved' && startsWith( github.repository, 'elementor/' )
```

**Review by ronkelementor — CHANGES_REQUESTED** [review]

File: .github/scripts/testim_ci.js (hunk 1 above)
Comment (maorZelinger) [reply .github/scripts/testim_ci.js]: Done

File: .github/workflows/testim-ci.yml (hunk 3 above)
Comment (maorZelinger) [reply .github/workflows/testim-ci.yml]: Removed

**Review by ronkelementor — DISMISSED** [review]

**Comment by ronkelementor** [comment]
@maorZelinger please change log print
image

**Review by ronkelementor — DISMISSED** [review]

**Comment by arielk** [comment]
@matipojo @KingYes is this PR still relevant?

**Pull request closed without merging**